options/db_options.cc

```diff
@@ -869,6 +869,13 @@ Status GetMutableDBOptionsFromStrings(
   if (!s.ok()) {
     *new_options = base_options;
   }
+  // check if new_options is changed by comparing to base_options
+  if (is_changed) {
+    std::string mismatch;
+    *is_changed = !OptionTypeInfo::StructsAreEqual(
+        config_options, "MutableDBOptions", &db_mutable_options_type_info,
+        "MutableDBOptions", &base_options, new_options, &mismatch);
+  }
```

Comment:
I think this would be better as a new API (AreMutableDBOptionsEqual ?).  This way, the DB code could do whatever sanitization is necessary before the comparison takes place.  Effectively the same code, just in a new routine.

Reply:
make sense.